Update docs and release notes with improved FIPS support (#7711)
strimzi/strimzi-kafka-operator: documentation/modules/operators/proc-disabling-fips-mode-cluster-operator.adoc

// Module included in the following assemblies:
//
// assembly-using-the-cluster-operator.adoc

[id='proc-disabling-fips-mode-cluster-operator-{context}']
= Disabling FIPS mode

[role="_abstract"]

> If you want, you can also disable the FIPS mode by setting the `FIPS_MODE` environment variable to `disabled` in the deployment configuration for the Cluster Operator.

Reviewer: ```suggestion
Strimzi automatically switches to FIPS mode when running on a FIPS-enabled Kubernetes cluster.
Disable FIPS mode by setting the `FIPS_MODE` environment variable to `disabled` in the deployment configuration for the Cluster Operator.
```

Reviewer: Might be useful to say why you might want to do this?

Author: > Might be useful to say why you might want to do this?

To be honest, I do not know. But I want to keep the option _just in case_ and since the doc is already there, though tit might be easiest to keep it 🤷

> With disabled FIPS mode, Strimzi will automatically disable FIPS in the OPenJDK in all components.

Reviewer: ```suggestion
With FIPS mode disabled, Strimzi automatically disables FIPS in the OpenJDK for all components.
```

> Such Strimzi deployment won't be FIPS compliant, and the Strimzi operators as well as all of its operands will be running in the same way as if they were running on an Kubernetes cluster without FIPS enabled.

Reviewer: ```suggestion
With FIPS mode disabled, Strimzi is not FIPS compliant.
The Strimzi operators, as well as all operands, run in the same way as if they were running on an Kubernetes cluster without FIPS enabled.
```
